lib/browser/api/navigation-controller.js

```diff
@@ -62,6 +62,25 @@ var NavigationController = (function () {
       options = {}
     }
     this.pendingIndex = -1
+    if (options.hasOwnProperty('extraHeaders')) {
+      let extraHeaders = options['extraHeaders']
+      if (typeof extraHeaders === 'object') {
```

Comment:
The `typeof` value for arrays is `'object'` so I'm not sure this is working correctly for the array case.